vpr/src/place/analytical_placement/analytical_placement_flow.cpp

```diff
@@ -51,26 +53,19 @@ void run_analytical_placement_flow() {
     // Using the placement from the SA to get the fixed block locations
     std::set<AtomBlockId> fixed_blocks;
     std::map<AtomBlockId, double> fixed_blocks_x, fixed_blocks_y;
+    const VprConstraints constraints = g_vpr_ctx.mutable_floorplanning().constraints;
     for (const AtomNetId& net_id : atom_netlist.nets()) {
         if (PartialPlacement::net_is_ignored_for_placement(atom_netlist, net_id))
             continue;
         for (AtomPinId pin_id : atom_netlist.net_pins(net_id)) {
             AtomBlockId blk_id = atom_netlist.pin_block(pin_id);
-            AtomBlockType blk_type = atom_netlist.block_type(blk_id);
-            if (blk_type == AtomBlockType::INPAD || blk_type == AtomBlockType::OUTPAD)
+            if (constraints.get_atom_partition(blk_id) != PartitionId::INVALID() && 
+                fixed_blocks.find(blk_id) == fixed_blocks.end()) {
```

Comment:
I like this check to ensure that the block is not already in fixed blocks. But if that is the case, we should probably check that it is trying to fix the block to the same location.

This would check for the case that the user accidentally fixed the same atom to two different regions. I am not sure if that is possible but it may be worth the double check.

Reply:
In this part of the code, I only call `constraints.get_atom_partition(blk_id)`. This function does a find from an existing map. It is the loading of the constraint file that should check for whether an atom is assigned to two different places. I have tried to create that error in .xml, vpr did not react, I think it would just store the latter region.

Reply:
Ah I see! Sounds good! I am happy that you checked. I like to know, and check, all corner cases since sometimes some really weird bugs can occur and small things like this tend to be the culprit.